✨ Add Firestore  🔥

RafaelZasas opened the pull request:
➕ firestore.js ✨ database logic:
getProducts() queries Firestore then calls renderData() to populate the UI

Accidentally left test code in profile.js

Updated <head> and <body> tags for each product page.

RafaelZasas committed:
✨ Add Firestore  🔥

➕ firestore.js ✨ database logic:
getProducts() queries Firestore then calls renderData() to populate the UI

ANDREYDEN commented on public/src/js/firestore.js:
```diff
@@ -0,0 +1,47 @@
+
+        querySnapshot.docs.forEach((doc) => {
+            // todo: Figure out if should use this or keep original data structure from firestore
+            products[doc.id] = {
+                "brand": doc.data().brand,
+                "price": doc.data().price,
+                "name": doc.data().name,
+                "onSale": doc.data().onSale
+            }
```
Possibly replace with
```suggestion
            products[doc.id] = { ...doc.data() }
```

RafaelZasas replied:
Didn't know this was possible. Thanks!

ANDREYDEN commented on public/src/js/profile.js:
```diff
@@ -6,3 +6,11 @@ function populateProfile() {
+
+function getFirestoreData(){
+    var output = document.getElementById('forData');
+    output.innerText = "test";
+    console.log("test")
```
Is this testing code or is it intentional?

RafaelZasas replied:
@ANDREYDEN its testing code- please se commit message

RafaelZasas committed:
⚡️ ♻️ Dynamically populate products grid

📱 Products section now automatically wraps after 4 cards. 

Changed renderData() to populateProductCards()

ANDREYDEN commented on public/src/js/firestore.js:
```diff
@@ -0,0 +1,58 @@
+
+    db.collection(`products/${productType}/inventory`).get().then((querySnapshot) => {
+
+        querySnapshot.docs.forEach((doc) => {
+            
+            
+            // todo: determine if we even need a locally stored map of documents
+            products[doc.id] = {
+                ...doc.data() // this replaces the need for individual assignments such as "brand": doc.data().brand,
+            }
```
I gave it a thought and this could be simplified even further:
Given that `products` is a simple array, then you can do something like this:
```
const products = querySnapshot.docs.map(doc => doc.data())
```
or
```
const products = querySnapshot.docs.map(doc => ({ ...doc.data(), doc.id }))
```

ANDREYDEN commented on public/src/js/firestore.js:
```diff
@@ -0,0 +1,58 @@
+                ...doc.data() // this replaces the need for individual assignments such as "brand": doc.data().brand,
+            }
+
+            populateProductCards(doc, index); // add each card individually; index is used to reference each of the cards
+            index++;
+        });
+
+        productsSection.innerHTML += productsHTML; // adds all of the cards html to the products grid
```
Also I would decouple the rendering part from Firestore queries. So maybe just have a separate function for fetching data and a separate one for rendering, which you would call on load.
